spyder/plugins/remoteclient/tests/conftest.py

```diff
@@ -0,0 +1,235 @@
+from spyder.plugins.remoteclient.api.client import SpyderRemoteClient
+from spyder.plugins.remoteclient.plugin import RemoteClient
+from spyder.plugins.remoteclient.widgets import AuthenticationMethod
+from spyder.utils.qthelpers import qapplication
+
+
+T = typing.TypeVar("T")
+
+# NOTE: These credentials are hardcoded in the Dockerfile.
+USERNAME = "ubuntu"
+PASSWORD = USERNAME
+
+HERE = Path(__file__).resolve().parent
+
+
+class MainWindowMock(QMainWindow):
+    """QMainWindow mock for the Remote Client plugin tests."""
+
+    def __init__(self):
+        # This avoids using the cli options passed to pytest
+        sys_argv = [sys.argv[0]]
+        self._cli_options = get_options(sys_argv)[0]
+        super().__init__()
+        PLUGIN_REGISTRY.set_main(self)
+
+    def register_plugin(self, plugin_class):
+        plugin = PLUGIN_REGISTRY.register_plugin(self, plugin_class)
+        plugin._register()
+        return plugin
+
+    @staticmethod
+    def unregister_plugin(plugin):
+        assert PLUGIN_REGISTRY.delete_plugin(
+            plugin.NAME
+        ), f"{plugin.NAME} not deleted"
+        plugin._unregister()
+
+    @staticmethod
+    def get_plugin(plugin_name, error=False):
+        return PLUGIN_REGISTRY.get_plugin(plugin_name)
+
+    @staticmethod
+    def is_plugin_available(plugin_name):
+        return PLUGIN_REGISTRY.is_plugin_available(plugin_name)
+
+
+@pytest.fixture(scope="session")
+def docker_compose_file(pytestconfig):
+    """Override the default docker-compose.yml file."""
+    return str(HERE / "docker-compose.yml")
+
+
+@pytest.fixture(scope="session")
+def remote_client_id(
+    ssh_server_addr: typing.Tuple[str, int], remote_client: RemoteClient
+) -> typing.Iterator[SpyderRemoteClient]:
+    """Add the Spyder Remote Client plugin to the registry."""
+    ssh_options = {
+        "host": ssh_server_addr[0],
+        "port": ssh_server_addr[1],
+        "username": USERNAME,
+        "password": PASSWORD,
+        "platform": "linux",
+        "known_hosts": None,
+    }
+    config_id = str(uuid.uuid4())
+
+    # Options Required by container widget
+    remote_client.set_conf(
+        f"{config_id}/auth_method", AuthenticationMethod.Password
+    )
+    remote_client.set_conf(
+        f"{config_id}/{AuthenticationMethod.Password}/name", "test-server"
+    )
+    remote_client.set_conf(
+        f"{config_id}/{AuthenticationMethod.Password}/address",
+        ssh_server_addr[0],
+    )
+    remote_client.set_conf(
+        f"{config_id}/{AuthenticationMethod.Password}/username", USERNAME
+    )
+
+    try:
+        remote_client.load_client(options=ssh_options, config_id=config_id)
+        yield config_id
+
+    finally:
+        remote_client.on_close(cancellable=False)
+
+
+@pytest.fixture(scope="session")
+def remote_client(
+    ipyconsole_and_remoteclient: typing.Tuple[IPythonConsole, RemoteClient],
+) -> RemoteClient:
+    """Start the Spyder Remote Client plugin.
```

Comment:
```suggestion
    """
    Start the Spyder Remote Client plugin.
```

This is our format for docstrings.